src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,162 @@
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUser(Long userId) {
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                return Optional.of(extractShoppingCartFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get shopping cart from DB with user ID = "
+                    + userId, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO shopping_carts(user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, shoppingCart.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+            return shoppingCart;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create shopping cart - " + shoppingCart, e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts sc\n"
+                + "INNER JOIN shopping_carts_products scp ON sc.cart_id = scp.cart_id\n"
+                + "INNER JOIN products p ON scp.product_id = p.product_id\n"
+                + "WHERE sc.cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            if (rs.next()) {
+                return Optional.of(extractShoppingCartFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get shopping cart from DB with ID = "
+                    + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts sc\n"
+                + "INNER JOIN shopping_carts_products scp ON sc.cart_id = scp.cart_id\n"
+                + "INNER JOIN products p ON scp.product_id = p.product_id\n";
```

Comment:
Why do you need JOIN here? Below you don't use any fields from joined tables - only cart_id and user_id from shopping_carts table.

Reply:
Agree.